### PR TITLE
Add tutorial: NFT Minter tutorial

### PR DESCRIPTION
**Pull request by nstrike2**
As a Go-To-Market engineer at Alchemy, I am creating a tutorial on how to create a full-stack NFT Minter due to heavily popular demand as a PR to the Ethereum foundation, titled "📝 NFT Minter Tutorial: How to Create a Full Stack DApp".

Added header tags for each of the titles as per request from @minimalsm.

Thank you and let's go builders!!!

## Description

As a Go-To-Market engineer at Alchemy, I am creating a tutorial on how to create a full-stack NFT Minter due to heavily popular demand as a PR to the Ethereum foundation, titled "📝 NFT Minter Tutorial: How to Create a Full Stack DApp".

## Related Issue

Continuation of https://github.com/ethereum/ethereum-org-website/pull/4162, which has now been closed due to inaccessibility.

https://github.com/ethereum/ethereum-org-website/issues/4134

### CODE REVIEW

**Comment by nstrike2** [comment]
@minimalsm @corwintines @pettinarip @samajammin @wackerow Can we get this merged? If not, happy to change what needs to be changed to get this merged.

**Comment by samajammin** [comment]
Thanks for the nudge - @corwintines or @minimalsm can you please review?

**Review by minimalsm — CHANGES_REQUESTED** [review]
Hey @nstrike2 thanks for the PR and sorry for the delay on this 😬🙏. I've left a quick first review to get the ball rolling ⚽️

I've left comments on the article itself but these are some general comments that are consistent across all of the tutorials you've posted. If we could resolve these we can get moving on these a lot faster.

- All headings must have header-ids
- Ethereum.org uses sentence casing for header names as a convention. Only the first word and proper nouns are capitalized. This applies to all markdown files on lines that begin with hashes (#).
- We can't use Emojis in headers as it causes issues with the table of contents
- We don't want a custom table of content's written into the content itself as we already have a sidebar component that handles this for us automatically
- Emojis should use the emoji component as [per our style guide](https://ethereum.org/en/contributing/style-guide/#images) to ensure consistency across devices and browsers
- All images must be saved locally (don't point to other websites). When adding an image to a page, the image should be downloaded and placed in the same folder as the markdown file. You can reference the image like this:`alt text for image`

File: src/content/developers/tutorials/nft-minter/index.md [hunk 1]
```diff
@@ -0,0 +1,931 @@
...
+
+![We will work inside the &quot;src&quot; folder](https://static.slab.com/prod/uploads/7adb25ff/posts/images/RQXRr70mP174PjsKQI_jMKuq.png)
+
+All of the code we'll write will live under the `src` folder. We'll be editing the `Minter.js` component and writing additional javascript files to give our project Web3 functionality.
+
+## Step 2: Check out our starter files {#step-2-check-out-our-starter-files}
+
+Before we start coding, it's important to check out what's already provided for us in the starter files.
+
+### Get your react project running
```
[the remark this answers is not in the document]
Comment (nstrike2) [reply src/content/developers/tutorials/nft-minter/index.md:77]: @minimalsm I don't need a header for a sub-bullet "Get your react project running" — I'm only using them for steps (e.g. Step 1, Step 2, etc.)

**Comment by nstrike2** [comment]
@minimalsm , at this point I've resolved all changes for this tutorial, and the hello world series (part 1/4, 2/4, 3/4, and 4/4). I hope we can close the loop here on all of them! Please let me know, as I'd love to get these all merged before the holidays.

**Comment by nstrike2** [comment]
@minimalsm can we merge this? I'll be leaving Alchemy in 3 days to go back to school and then can't do much from here to continue on this after the holidays, so would love to have these be mutually beneficial to both you and us! Have made all the changes — should be ready to go.

File: src/content/developers/tutorials/nft-minter/index.md [hunk 2]
```diff
@@ -0,0 +1,887 @@
...
+
+In this tutorial, we will be using [React](https://reactjs.org/) as our frontend framework. Because this tutorial is primarily focused on Web3 development, we won't be spending much time breaking down React fundamentals. Instead, we'll be focusing on bringing functionality to our project.
+
+As a prerequisite, you should have a beginner-level understanding of React—know how components, props, useState/useEffect, and basic function calling works. If you've never heard of any of those terms before, you may want to check out this [Intro to React tutorial](https://reactjs.org/tutorial/tutorial.html). For the more visual learners, we highly recommend this excellent [Full Modern React Tutorial](https://www.youtube.com/playlist?list=PL4cUxeGkcC9gZD-Tvwfod2gaISzfRiP9d) video series by Net Ninja.
+
+And if you haven't already, you'll definitely need an Alchemy account to complete this tutorial as well as build anything on the blockchain. Sign up for a free account [here](https://alchemy.com/)!
+
+Without further ado, let's get started!
+
+## Step 0: Making NFTs 101 {#making-nfts-101}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:40]:
```suggestion
## Making NFTs 101 {#making-nfts-101}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 3]
```diff
@@ -0,0 +1,887 @@
...
+
+Before we even start looking at any code, it's important to understand how making an NFT works. It involves two steps:
+
+### 1\) You publish an NFT smart contract on the Ethereum blockchain. Usually this is an [ERC-721](https://eips.ethereum.org/EIPS/eip-721) or [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155) smart contract.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:44]:
```suggestion
### Publish an NFT smart contract on the Ethereum blockchain {#publish-nft}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 4]
```diff
@@ -0,0 +1,887 @@
...
+
+The biggest difference between the two NFT smart contact standards is that ERC-1155 is a multi-token standard and includes batch functionality, whereas with the ERC-721 is a single-token standard and therefore only supports transferring one token at a time.
+
+### 2\) You call the minting function on that NFT smart contract to mint the NFT. _Minting_ is simply the act of publishing a unique instance of your non fungible token on the blockchain.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:48]:
```suggestion
### Call the minting function {#minting-function}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 5]
```diff
@@ -0,0 +1,887 @@
...
+npm install
+```
+
+Once those have finished installing,  run `npm start` in your terminal:
+
+```bash
+npm start
+```
+
+Doing so should open [http://localhost:3000/](http://localhost:3000/) in your browser, where you'll see the frontend for our project. It should consist of 3 fields: a place to input a link to your NFT's asset, enter the name of your NFT, and provide a description.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:94]:
```suggestion
Doing so should open http://localhost:3000/ in your browser, where you'll see the frontend for our project. It should consist of 3 fields: a place to input a link to your NFT's asset, enter the name of your NFT, and provide a description.
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 6]
```diff
@@ -0,0 +1,887 @@
...
+
+If you try clicking "Connect Wallet" or "Mint NFT" buttons, you'll notice they don't work—that's because we still need to program their functionality! :\)
+
+### The `Minter.js` component
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:98]:
```suggestion
### The Minter.js component {#minter-js}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 7]
```diff
@@ -0,0 +1,887 @@
...
+
+**In this tutorial, we'll only be editing the `Minter.js file` and adding files in our `src` folder.** 
+
+Now that we understand what we're working with, let's set up our Ethereum wallet!
+
+## Step 3: Set up your Ethereum wallet {#step-3-set-up-your-ethereum-wallet}
+
+For users to be able to interact with your smart contract they will need to connect their Ethereum wallet to your dApp.  
+
+### Download Metamask
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:211]:
```suggestion
### Download Metamask {#download-metamask}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 8]
```diff
@@ -0,0 +1,887 @@
...
+
+If you have Metamask installed, you should be prompted to connect your wallet to your dApp. Accept the invitation to connect.
+
+You should see that the wallet button now reflects that your address is connected! Yasssss 🔥
+
+Next, try refreshing the page... this is strange. Our wallet button is prompting us to connect Metamask, even though it is already connected...
+
+Don't worry though! We easily can fix that by implementing a function called `getCurrentWalletConnected`, which will check if an address is already connected to our dApp and update our UI accordingly!  
+
+### The `getCurrentWalletConnected` function
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:348]:
```suggestion
### The getCurrentWalletConnected function {#get-current-wallet}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 9]
```diff
@@ -0,0 +1,887 @@
...
+    setWallet(address)
+    setStatus(status);
+    
+    addWalletListener(); 
+}, []);
+```
+
+And voila! We've  completed programming all of our wallet functionality! Now that our wallet is set up, let's figure out how to mint our NFT!
+
+## Step 5: NFT Metadata 101 {#nft-metadata-101}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:479]:
```suggestion
## NFT Metadata 101 {#nft-metadata-101}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 10]
```diff
@@ -0,0 +1,887 @@
...
+
+So remember the NFT metadata we just talked about in Step 0 of this tutorial—it brings an NFT to life, allowing it to have properties, such as a digital asset, name, description, and other attributes.
+
+We're going to need to configure this metadata as a JSON object and store it, so we can pass it in as the `tokenURI` parameter when calling our smart contract's `mintNFT` function.
+
+The text in the "Link to Asset", "Name", "Description" fields will comprise the different properties of our NFT's metadata. We'll format this metadata as a JSON object, but there are a couple options for where we can store this JSON object:
+
+* We could store it on the Ethereum blockchain; however, doing so would be SUPER expensive \(we're talking upwards of hundreds of dollars\) due to the nature of Ethereum. ❌
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:487]:
```suggestion
- We could store it on the Ethereum blockchain; however, doing so would be very expensive.
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 11]
```diff
@@ -0,0 +1,887 @@
...
+* We could store it on a centralized server, like AWS or Firebase. But that would defeat our decentralization ethos. ❌
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:488]:
```suggestion
- We could store it on a centralized server, like AWS or Firebase. But that would defeat our decentralization ethos.
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 12]
```diff
@@ -0,0 +1,887 @@
...
+* We could use IPFS, a decentralized protocol and peer-to-peer network for storing and sharing data in a distributed file system. As this protocol is decentralized and free, it is our best option! ✅
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:489]:
```suggestion
- We could use IPFS, a decentralized protocol and peer-to-peer network for storing and sharing data in a distributed file system. As this protocol is decentralized and free, it is our best option!
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 13]
```diff
@@ -0,0 +1,887 @@
...
+
+To store our metadata on IPFS, we will use [Pinata](https://pinata.cloud/), a convenient IPFS API and toolkit. In the next step, we'll explain exactly how to do this!
+
+## Step 6: Use Pintata to pin your metadata to IPFS {#step-6-use-pinata-to-pin-your-metadata-to-IPFS}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:493]:
```suggestion
## Use Pintata to pin your metadata to IPFS {#use-pinata-to-pin-your-metadata-to-IPFS}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 14]
```diff
@@ -0,0 +1,887 @@
...
+
+If you don't have a [Pinata](https://pinata.cloud/) account, sign up for a free account [here](https://pinata.cloud/signup) and complete the steps to verify your email and account.
+
+### Create your Pinata API key
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:497]:
```suggestion
### Create your Pinata API key {#create-pinata-api-key}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 15]
```diff
@@ -0,0 +1,887 @@
...
+Next, in VSCode, navigate to your `.env` file and add your Pinata API key and API secret to it, like so:
+
+```text
+REACT_APP_PINATA_KEY = <pinata-api-key>
+REACT_APP_PINATA_SECRET = <pinata-api-secret>
+```
+
+Save the file, and then you're ready to start writing the function to upload your JSON metadata to IPFS!
+
+### Implement `pinJSONToIPFS`
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:532]:
```suggestion
### Implement pinJSONToIPFS {#pin-json-to-ipfs}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 16]
```diff
@@ -0,0 +1,887 @@
...
+};
+```
+
+## Step 10: Deploy your NFT to a live website {#deploy-your-NFT}
+
+Ready to take your project live for users to interact with? Check out [this tutorial](https://docs.alchemy.com/alchemy/tutorials/nft-minter/how-do-i-deploy-nfts-online) for deploying your Minter to a live website.
+
+One last step...
+
+## Step 11: Take the blockchain world by storm {#take-the-blockchain-world-by-storm}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:869]:
```suggestion
## Take the blockchain world by storm {#take-the-blockchain-world-by-storm}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 17]
```diff
@@ -0,0 +1,887 @@
...
+Finally, implement the `onMintPressed` function to make await call to your imported `mintNFT`function and update the `status` state variable to reflect whether our transaction succeeded or failed:
+
+```javascript
+const onMintPressed = async () => {
+    const { status } = await mintNFT(url, name, description);
+    setStatus(status);
+};
+```
+
+## Step 10: Deploy your NFT to a live website {#deploy-your-NFT}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:863]:
```suggestion
## Deploy your NFT to a live website {#deploy-your-NFT}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 18]
```diff
@@ -0,0 +1,887 @@
...
+
+Presumably, you'd like to be able to show off the NFTs minted via your dApp in your wallet — so be sure to check out our quick tutorial [How to View Your NFT in Your Wallet](https://docs.alchemyapi.io/alchemy/tutorials/how-to-write-and-deploy-a-nft-smart-contract/how-to-view-your-nft-in-your-wallet)!
+
+And, as always, if you have any questions, we're here to help in the [Alchemy Discord](https://discord.gg/gWuC7zB). We can't wait to see how you apply the concepts from this tutorial to your future projects!  🧙‍♂️
+
+## Bonus: Put your NFT Minter to work!
+
+Ready to mint an NFT using your minter? Check out this [demo video](https://youtu.be/mmrO6_5-hRU)!
+
+**NOTE:** The transactions sent through Metamask will use Metamask's default provider \(not necessarily your Alchemy endpoint\). If you'd like to send those transactions through Alchemy to unlock debugging tools and the [Alchemy Notify ](../../introduction/core-products/alchemy-notify.md)suite, check out [this guide](../../guides/connecting-metamask-to-alchemy.md) for connecting Metamask to Alchemy.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:885]:
```suggestion
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 19]
```diff
@@ -0,0 +1,887 @@
...
+
+JK, you made it to the end of the tutorial! To recap, by building an NFT minter, you successfully learned how to:
+
+* Connect to Metamask via your frontend project
+* Call smart contract methods from your frontend
+* Sign transactions using Metamask
+
+Presumably, you'd like to be able to show off the NFTs minted via your dApp in your wallet — so be sure to check out our quick tutorial [How to View Your NFT in Your Wallet](https://docs.alchemyapi.io/alchemy/tutorials/how-to-write-and-deploy-a-nft-smart-contract/how-to-view-your-nft-in-your-wallet)!
+
+And, as always, if you have any questions, we're here to help in the [Alchemy Discord](https://discord.gg/gWuC7zB). We can't wait to see how you apply the concepts from this tutorial to your future projects!  🧙‍♂️
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:879]:
```suggestion
And, as always, if you have any questions, we're here to help in the [Alchemy Discord](https://discord.gg/gWuC7zB). We can't wait to see how you apply the concepts from this tutorial to your future projects! 
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 20]
```diff
@@ -0,0 +1,887 @@
...
+
+## Bonus: Put your NFT Minter to work!
+
+Ready to mint an NFT using your minter? Check out this [demo video](https://youtu.be/mmrO6_5-hRU)!
+
+**NOTE:** The transactions sent through Metamask will use Metamask's default provider \(not necessarily your Alchemy endpoint\). If you'd like to send those transactions through Alchemy to unlock debugging tools and the [Alchemy Notify ](../../introduction/core-products/alchemy-notify.md)suite, check out [this guide](../../guides/connecting-metamask-to-alchemy.md) for connecting Metamask to Alchemy. 
+
+Once you complete this tutorial, let us know how your experience was or if you have any feedback by tagging us on Twitter [@alchemyplatform](https://twitter.com/AlchemyPlatform)!
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:887]:
```suggestion
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 21]
```diff
@@ -0,0 +1,887 @@
...
+            success: false,
+            status: "😥 Something went wrong: " + error.message
+        }
+    }
+}
+```
+
+That's one giant function! Now, we just need to connect our `mintNFT` function to our `Minter.js` component...
+
+## Step 9: Connect mintNFT to our Minter.js frontend {#connect-our-frontend}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:846]:
```suggestion
## Connect mintNFT to our Minter.js frontend {#connect-our-frontend}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 22]
```diff
@@ -0,0 +1,887 @@
...
+const { createAlchemyWeb3 } = require("@alch/alchemy-web3");
+const web3 = createAlchemyWeb3(alchemyKey); 
+
+const contractABI = require('../contract-abi.json')
+const contractAddress = "0x4C4a07F737Bf57F6632B6CAB089B78f62385aCaE";
+```
+
+Once we have both of those, we're ready to start coding our mint function!
+
+## Step 8: Implement the mintNFT function {#step-8-implement-the-mintnft-function}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:661]:
```suggestion
## Implement the mintNFT function {#implement-the-mintnft-function}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 23]
```diff
@@ -0,0 +1,887 @@
...
+
+### Create your Alchemy API key
+
+If you don't already have an Alchemy account, [sign up for free here.](https://alchemy.com/?a=eth-org-nft-minter)
+
+Once you’ve created an Alchemy account, you can generate an API key by creating an app. This will allow us to make requests to the Ropsten test network. 
+
+Navigate to the “Create App” page in your Alchemy Dashboard by hovering over “Apps” in the nav bar and clicking “Create App”.
+
+Name your app \(we chose "My First NFT!"\), offer a short description, select “Staging” for the Environment \(used for your app bookkeeping\), and choose “Ropsten” for your network.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:611]:
```suggestion
Name your app we chose "My First NFT!", offer a short description, select “Staging” for the Environment used for your app bookkeeping, and choose “Ropsten” for your network.
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 24]
```diff
@@ -0,0 +1,887 @@
...
+
+As with our `connectWallet`function return types, we're returning JSON objects so we can use  their parameters to update our state variables and UI.
+
+## Step 7: Load your smart contract {#load-your-smart-contract}
+
+Now that we have a way to upload our NFT metadata to IPFS via our `pinJSONToIPFS` function, we're going to need a way to load an instance of our smart contract so we can call its `mintNFT` function.
+
+As we mentioned earlier, in this tutorial we will be using [this existing NFT smart contract](https://ropsten.etherscan.io/address/0x4C4a07F737Bf57F6632B6CAB089B78f62385aCaE); however, if you'd like to learn how we made it, or make one yourself, we highly recommend you check out our other tutorial, ["How to Create an NFT."](https://docs.alchemyapi.io/alchemy/tutorials/how-to-create-an-nft).
+
+### The contract ABI
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:597]:
```suggestion
### The contract ABI {#contract-abi}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 25]
```diff
@@ -0,0 +1,887 @@
...
+First, it imports [axios](https://www.npmjs.com/package/axios), a promise based HTTP client for the browser and node.js, which we will use to make a request to Pinata.
+
+Then we have our asynchronous function `pinJSONToIPFS`, which takes a `JSONBody` as its input and the Pinata api key and secret in its header, all to make a POST request to their `pinJSONToIPFS` API.
+
+* If this POST request is successful, then our function returns a JSON object with the `success` boolean as true and the `pinataUrl` where our metadata was pinned. We will use this `pinataUrl` returned as the `tokenURI` input to our smart contract's mint function.
+* If this post request fails, then our function returns a JSON object with the `success` boolean as false and a `message` string that relays our error.
+
+As with our `connectWallet`function return types, we're returning JSON objects so we can use  their parameters to update our state variables and UI.
+
+## Step 7: Load your smart contract {#load-your-smart-contract}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:591]:
```suggestion
## Load your smart contract {#load-your-smart-contract}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 26]
```diff
@@ -0,0 +1,887 @@
...
+useEffect(async () => {
+    const {address, status} = await getCurrentWalletConnected();
+    setWallet(address)
+    setStatus(status); 
+}, []);
+```
+
+Notice, we use the response of our call to `getCurrentWalletConnected` to update our `walletAddress` and `status` state variables. 
+
+Once you've added this code, try refreshing our browser window. The button should say that you're connected, and show a preview of your connected wallet's address - even after you refresh! 😅
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:424]:
```suggestion
Once you've added this code, try refreshing our browser window. The button should say that you're connected, and show a preview of your connected wallet's address - even after you refresh! 
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 27]
```diff
@@ -0,0 +1,887 @@
...
+
+In `connectWalletPressed`, we simply make an await call to our imported `connectWallet` function, and using its response, we update our `status` and `walletAddress` variables via their state hooks.
+
+Now, let's save both files \(`Minter.js` and `interact.js`\) and test out our UI so far. 
+
+Open your browser on the [http://localhost:3000/](http://localhost:3000/) page, and press the "Connect Wallet" button on the top right of the page.
+
+If you have Metamask installed, you should be prompted to connect your wallet to your dApp. Accept the invitation to connect.
+
+You should see that the wallet button now reflects that your address is connected! Yasssss 🔥
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:342]:
```suggestion
You should see that the wallet button now reflects that your address is connected.
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 28]
```diff
@@ -0,0 +1,887 @@
...
+    setStatus(walletResponse.status);
+    setWallet(walletResponse.address);
+  };
+```
+
+Notice how most of our functionality is abstracted away from our `Minter.js` component from the `interact.js` file? This is so we comply with the M-V-C paradigm!
+
+In `connectWalletPressed`, we simply make an await call to our imported `connectWallet` function, and using its response, we update our `status` and `walletAddress` variables via their state hooks.
+
+Now, let's save both files \(`Minter.js` and `interact.js`\) and test out our UI so far.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:336]:
```suggestion
Now, let's save both files `Minter.js` and `interact.js` and test out our UI so far. 
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 29]
```diff
@@ -0,0 +1,887 @@
...
+  };
+```
+
+Notice how most of our functionality is abstracted away from our `Minter.js` component from the `interact.js` file? This is so we comply with the M-V-C paradigm!
+
+In `connectWalletPressed`, we simply make an await call to our imported `connectWallet` function, and using its response, we update our `status` and `walletAddress` variables via their state hooks.
+
+Now, let's save both files \(`Minter.js` and `interact.js`\) and test out our UI so far. 
+
+Open your browser on the [http://localhost:3000/](http://localhost:3000/) page, and press the "Connect Wallet" button on the top right of the page.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:338]:
```suggestion
Open your browser on the http://localhost:3000/ page, and press the "Connect Wallet" button on the top right of the page.
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 30]
```diff
@@ -0,0 +1,887 @@
...
+
+Phew! Our fake money is all there! <Emoji text=":money_mouth_face:" size={1} />
+
+## Step 4: Connect Metamask to your UI {#connect-metamask-to-your-UI}
+
+Now that our Metamask wallet is set up, let's connect our dApp to it!
+
+Because we want to prescribe to the [M-V-C](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) paradigm, we're going to create a separate file that contains our functions to manage the logic, data, and rules of our dApp, and then pass those functions to our frontend \(our Minter.js component\).
+
+### The `connectWallet` function
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:239]:
```suggestion
### The `connectWallet` function {#connect-wallet-function}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 31]
```diff
@@ -0,0 +1,887 @@
...
+
+```text
+{"jsonrpc": "2.0", "id": 0, "result": "0xde0b6b3a7640000"}
+```
+
+**NOTE:** This result is in wei not eth. Wei is used as the smallest denomination of ether. The conversion from wei to eth is: 1 eth = 10¹⁸ wei. So if we convert 0xde0b6b3a7640000 to decimal we get 1\*10¹⁸ which equals 1 eth.
+
+Phew! Our fake money is all there! <Emoji text=":money_mouth_face:" size={1} />
+
+## Step 4: Connect Metamask to your UI {#connect-metamask-to-your-UI}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:233]:
```suggestion
## Connect Metamask to your UI {#connect-metamask-to-your-UI}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 32]
```diff
@@ -0,0 +1,887 @@
...
+
+For users to be able to interact with your smart contract they will need to connect their Ethereum wallet to your dApp.  
+
+### Download Metamask
+
+For this tutorial, we’ll use Metamask, a virtual wallet in the browser used to manage your Ethereum account address. If you want to understand more about how transactions on Ethereum work, check out [this page](/developers/docs/transactions/) from the Ethereum foundation.
+
+You can download and create a Metamask account for free [here](https://metamask.io/download.html). When you are creating an account, or if you already have an account, make sure to switch over to the “Ropsten Test Network” in the upper right \(so that we’re not dealing with real money\).
+
+### Add ether from a Faucet <a id="step-4-add-ether-from-a-faucet"></a>
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:217]:
```suggestion
### Add ether from a Faucet {#add-ether-from-faucet}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 33]
```diff
@@ -0,0 +1,887 @@
...
+
+Now that we understand what we're working with, let's set up our Ethereum wallet!
+
+## Step 3: Set up your Ethereum wallet {#step-3-set-up-your-ethereum-wallet}
+
+For users to be able to interact with your smart contract they will need to connect their Ethereum wallet to your dApp.  
+
+### Download Metamask
+
+For this tutorial, we’ll use Metamask, a virtual wallet in the browser used to manage your Ethereum account address. If you want to understand more about how transactions on Ethereum work, check out [this page](/developers/docs/transactions/) from the Ethereum foundation.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:213]:
```suggestion
For this tutorial, we’ll use Metamask, a virtual wallet in the browser used to manage your Ethereum account address. If you want to understand more about how transactions on Ethereum work, check out [this page](/developers/docs/transactions/).
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 34]
```diff
@@ -0,0 +1,887 @@
...
+
+Finally, let's address where is this Minter component added.
+
+If you go to the `App.js` file, which is the main component in React that acts as a container for all other components, you'll see that our Minter component is injected on line 7.
+
+**In this tutorial, we'll only be editing the `Minter.js file` and adding files in our `src` folder.** 
+
+Now that we understand what we're working with, let's set up our Ethereum wallet!
+
+## Step 3: Set up your Ethereum wallet {#step-3-set-up-your-ethereum-wallet}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:207]:
```suggestion
##: Set up your Ethereum wallet {#set-up-your-ethereum-wallet}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 35]
```diff
@@ -0,0 +1,887 @@
...
+
+An NFT's metadata is really what brings it to life, allowing it to have properties, such as a name, description, image \(or different digital asset\), and other attributes. Here's [an example of a tokenURI](https://gateway.pinata.cloud/ipfs/QmSvBcb4tjdFpajGJhbFAWeK3JAxCdNQLQtr6ZdiSi42V2), which contains an NFT's metadata.
+
+In this tutorial, we're going to focus on part 2, calling an existing NFT's smart contract minting function using our React UI. 
+
+[Here's a link](https://ropsten.etherscan.io/address/0x4C4a07F737Bf57F6632B6CAB089B78f62385aCaE) to the ERC-721 NFT smart contract we will be calling in this tutorial. If you'd like to learn how we made it, we highly recommend that you check out our other tutorial, ["How to Create an NFT"](https://docs.alchemyapi.io/alchemy/tutorials/how-to-create-an-nft).
+
+Cool, now that we understand how making an NFT works, let's clone our starter files!
+
+## Step 1: Clone the starter files {#clone-the-starter-files}
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:60]:
```suggestion
## Clone the starter files {#clone-the-starter-files}
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 36]
```diff
@@ -0,0 +1,887 @@
...
+    "UI",
+    "wallet",
+    "Pinata",
+  ]
+skill: intermediate
+lang: en
+sidebar: true
+published: 2021-10-06
+---
+# 📝 NFT Minter Tutorial: How to Create a Full Stack DApp
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:22]:
```suggestion
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 37]
```diff
@@ -0,0 +1,887 @@
...
+
+One of the greatest challenges for developers coming from a Web2 background is figuring out how to connect your smart contact to a frontend project and interact with it.
+
+By building an NFT minter — a simple UI where you can input a link to your digital asset, a title,  and a description — you'll learn how to:
+
+* Connect to Metamask via your frontend project
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:28]:
```suggestion
- Connect to Metamask via your frontend project
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 38]
```diff
@@ -0,0 +1,887 @@
...
+* Call smart contract methods from your frontend
+* Sign transactions using Metamask
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:30]:
```suggestion
- Call smart contract methods from your frontend
- Sign transactions using Metamask
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 39]
```diff
@@ -0,0 +1,887 @@
...
+
+Cool, now that we understand how making an NFT works, let's clone our starter files!
+
+## Step 1: Clone the starter files {#clone-the-starter-files}
+
+First, go to the [nft-minter-tutorial GitHub repository](https://github.com/alchemyplatform/nft-minter-tutorial) to get the starter files for this project. Clone this repository into your local environment.=
+
+When you open this cloned `nft-minter-tutorial` repository, you'll notice that it contains two folders: `minter-starter-files` and `nft-minter`.
+
+* `minter-starter-files` contains the starter files \(essentially the React UI\) for this project. In this tutorial, **we will be working in this directory**, as you learn how to bring this UI to life by connecting it to your Ethereum wallet and an NFT smart contract.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:66]:
```suggestion
* `minter-starter-files` contains the starter files (essentially the React UI) for this project. In this tutorial, **we will be working in this directory**, as you learn how to bring this UI to life by connecting it to your Ethereum wallet and an NFT smart contract.
```

File: src/content/developers/tutorials/nft-minter/index.md [hunk 40]
```diff
@@ -0,0 +1,887 @@
...
+* `nft-minter` contains the entire completed tutorial and is there for you as a **reference** **if you get stuck.**
+
+Next open your copy of `minter-starter-files` in to your favorite code editor \(at Alchemy, we we're big fans of [VSCode](https://code.visualstudio.com/download)\), and then navigate into your `src` folder.
```
Comment (minimalsm) [inline src/content/developers/tutorials/nft-minter/index.md:69]:
```suggestion
Next, open your copy of `minter-starter-files` in your code editor, and then navigate into your `src` folder.
```

**Review by minimalsm — CHANGES_REQUESTED** [review]

**Review by minimalsm — APPROVED** [review]

**Pull request merged**